### PR TITLE
[SPARK-12446][SQL] Add unit tests for JDBCRDD internal functions

### PR DESCRIPTION
**Pull request by maropu**
No tests done for JDBCRDD#compileFilter.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala [hunk 1]
```diff
@@ -168,6 +168,35 @@ private[sql] object JDBCRDD extends Logging {
   }
 
   /**
+   * Converts value to SQL expression.
+   */
+  private def compileValue(value: Any): Any = value match {
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: did you just move this code block, or was there any changes made?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala [hunk 2]
```diff
@@ -381,6 +382,8 @@ private[sql] class JDBCRDD(
     val myWhereClause = getWhereClause(part)
 
     val sqlText = s"SELECT $columnList FROM $fqTable $myWhereClause"
+    logDebug(s"'${sqlText}' input into JDBC")
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: remove this?

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala [hunk 3]
```diff
@@ -262,40 +291,12 @@ private[sql] class JDBCRDD(
...
-    case IsNotNull(attr) => s"$attr IS NOT NULL"
-    case _ => null
-  }
 
   /**
    * `filters`, but as a WHERE clause suitable for injection into a SQL query.
    */
   private val filterWhereClause: String = {
-    val filterStrings = filters map compileFilter filter (_ != null)
+    val filterStrings = filters map JDBCRDD.compileFilter filter (_ != null)
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: remove the usage of infix notations here

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala [hunk 4]
```diff
@@ -18,18 +18,23 @@
...
+import org.apache.spark.sql.execution.datasources.jdbc.JDBCRDD
 import org.apache.spark.sql.test.SharedSQLContext
 import org.apache.spark.sql.types._
+import org.apache.spark.sql.sources._
 import org.apache.spark.util.Utils
 
-class JDBCSuite extends SparkFunSuite with BeforeAndAfter with SharedSQLContext {
+class JDBCSuite extends SparkFunSuite
+  with BeforeAndAfter with PrivateMethodTester with SharedSQLContext
+{
```
Comment (rxin) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala]: put this on the last line?

**Comment by rxin** [comment]
Let me know once you address my comments. This looks pretty good, assuming you mostly just moved the code to become static functions.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala (hunk 1 above)
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: I just moved `compileValue`, `escapeSql`, and `compileFilter` in this companion object area.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala (hunk 3 above)
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: Okay

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala [hunk 5]
```diff
@@ -168,6 +168,35 @@ private[sql] object JDBCRDD extends Logging {
   }
 
   /**
+   * Converts value to SQL expression.
+   */
+  private def compileValue(value: Any): Any = value match {
+    case stringValue: String => s"'${escapeSql(stringValue)}'"
+    case timestampValue: Timestamp => "'" + timestampValue + "'"
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: @rxin `s"'$timestampValue'"`` is better here?

Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: both works - i wouldn't worry about it.

File: sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala (hunk 4 above)
Comment (maropu) [inline sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala]: Okay

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala (hunk 5 above)
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala]: Okay.

**Comment by maropu** [comment]
@rxin Finished.

**Comment by rxin** [comment]
LGTM. We should merge this as soon as tests pass.

**Comment by rxin** [comment]
Thanks - I've merged this.

**Pull request closed without merging**